pkg/kubernetes/deploy.js

```diff
@@ -73,6 +80,220 @@ define([
+
+                        var promise2 = nulecule_client.installrun("install", install_dir, fields.nulecule_image)
+                            .done(function() {
+
+                                var promise3 = nulecule_client.get_statuslist()
+                                    .done(function(ans_json){
+                                        answerfile = nulecule_client.loadAnswersfile(ans_json);
+
+                                        //Show App Params in the dialog
+                                        var template = $('#deploy-app-appentity-template').html();
+                                        Mustache.parse(template);
+                                        var text = Mustache.render(template, $.extend({
+                                            "apps": convertToKV(answerfile)
+                                        }));
+                                        $('#deploy-app-dialog').find('.cockpit-form-table').append(text);
+                                        
+                                        run_stage = true;
+                                        return;
+
+                                    })
+                                    .fail(function(ex, response) {
+                                        /* Display the error appropriately in the dialog */
+                                        $("#deploy-app-dialog").dialog("failure", ex);
+                                    });
+                            })
+                            .fail(function(ex, response) {                               
+                                /* Display the error appropriately in the dialog */
+                                $("#deploy-app-dialog").dialog("failure", ex);
+                            });
+
+
+                        /* Display a spinner while install is happening */
+                        $("#deploy-app-dialog").dialog("wait", promise2);
+                    })
+                    .fail(function(ex, response) {
+                        /* Display the error appropriately in the dialog */
+                        $("#deploy-app-dialog").dialog("failure", ex);
+                    });
+
+                    /* Display a spinner while tmp folder is happening */
+                    $("#deploy-app-dialog").dialog("wait", promise1);
+
+            });
+
+        /* Display a spinner while vaidation is happening */
+        $("#deploy-app-dialog").dialog("wait", promise);
+    }
+
+
+    function validate_app_fields() {
+        var dfd = $.Deferred();
+        var ex, fails = [];
+        var label_input = get_fields(convertToKV(answerfile));
+        var fields = { };
+        var tmp = nulecule_client.get_answers();
+
+        for(var x in label_input) {
+            var li = label_input[x];
+            var input_id = "#" + li.input_val;
+            var label_id = li.label_val;
+            var app_data = get_app_from_label(label_id);
+            var input_value = $(input_id).val();
```

Comment:
Declaring variables inside a loop is poor javascript form. Variables are function scoped. Suggest delcaling these outside the loop, and assigning them in here. Alternatively use $.each(label_input, function() { ... });